src/libsystemd/sd-device/device-monitor.c

```diff
@@ -245,6 +245,8 @@ static int device_monitor_event_handler(sd_event_source *s, int fd, uint32_t rev
         if (device_monitor_receive_device(m, &device) <= 0)
                 return 0;
 
+        LOG_CONTEXT_CONSUME_STRV(device_log_context_fields(device));
```

Comment:
Setting the log context here may be too early, as an uevent may trigger very long operation.

Reply:
I don't understand, this is right before we invoke the callback no? If this is too early, where would you suggest we set the log context for the device?

Reply:
In other words, the usage of log context I thought was that each sd-device object manages its log context for the device, and push and pop within single call of `log_device_xxx()`. Then, log messages unrelated to the device are not polluted by the log context.

But now, you propose all logs during a uevent attaches the log context about the received device. That might help in some case. But when we widely use the log context for variety of objects, then each log messages has toooo many fields...

Reply:
The device log context is only valid for the duration of the callback though. And I figure that during the callback for a specific device, it makes sense to attach the log context fields related to the device to every log message logged during the callback.

This makes it very easy to show all log messages related to the device by simply doing `journalctl DEVNAME=<devname>`.

The idea is that during a device monitor callback for a device X, even when we're calling into code that isn't aware of the device being processed, we can still attach the necessary context to those log messages to be able to figure out later that they were logged when device X was being processed.

To be able to only show the fields you're interested in, I've been working on https://github.com/systemd/systemd/pull/23635 which will allow to format the journal output to only show the fields the user is interested in.

As for disk space constraints, the journal deduplicates identical fields so logging the same fields many times for many log messages is cheap.